lib/auth/auth_xport_common.c

```diff
@@ -1003,6 +1004,19 @@ void auth_message_hdr_to_be16(struct auth_message_frag_hdr *frag_hdr)
 	frag_hdr->payload_len = sys_cpu_to_be16(frag_hdr->payload_len);
 }
 
+/**
+ * @see auth_internal.h
+ */
+void auth_get_random(uint8_t *buf, size_t num)
```

Comment:
New common Random function.

Reply:
If we did agree to create a new random function, I think it would be appropriate to add it as its own change. However, I would argue that if there isn't a cryptographically secure random source available, the security added by this code is really more of an illusion. Non cryptographic random sources are rather predictable, and generally result in catastrophic weaknesses when used with cryptograph protocols, such as this.

Reply:
I don't think you should be defining yet another wrapper around rand. Decide whether your subsystem requires a cryptographically secure random values. If it does then just call sys_csrand_get() directly because that shows the reader the intent. If it doesn't then call sys_rand_get(). This way, if the underlying HW doesn't support a cryptographically secure random and someone tries to enable code that requires it, a build error will occur.